test/pkg.jl

```diff
@@ -287,72 +264,46 @@ temp_pkg_dir() do
     end
 
     # Various pin/free/re-pin/change-pin patterns (issue #17176)
-    begin
-        ret, out, err = @grab_outputs Pkg.free("Example")
-        @test ret === nothing && out == ""
-        @test contains(err, "INFO: Freeing Example")
-
-        ret, out, err = @grab_outputs Pkg.pin("Example")
-        @test ret === nothing && out == ""
-        @test ismatch(r"INFO: Creating Example branch pinned\.[0-9a-f]{8}\.tmp", err)
-        @test !contains(err, "INFO: No packages to install, update or remove")
-        branchid = replace(err, r".*pinned\.([0-9a-f]{8})\.tmp.*"s, s"\1")
+    @test "" == capture_stdout() do
+        @test_warn "INFO: Freeing Example" Pkg.free("Example")
+
+        @test_warn r"^INFO: Creating Example branch pinned\.[0-9a-f]{8}\.tmp$" Pkg.pin("Example")
         vers = Pkg.installed("Example")
+        branch = LibGit2.with(LibGit2.GitRepo, Pkg.dir("Example")) do repo
+          LibGit2.branch(repo)
+       end
 
-        ret, out, err = @grab_outputs Pkg.free("Example")
-        @test ret === nothing && out == ""
-        @test contains(err, "INFO: Freeing Example")
+        @test_warn "INFO: Freeing Example" Pkg.free("Example")
 
-        ret, out, err = @grab_outputs Pkg.pin("Example", v"0.4.0")
-        @test ret === nothing && out == ""
-        @test contains(err, "INFO: Creating Example branch pinned.b1990792.tmp")
-        @test contains(err, "INFO: No packages to install, update or remove")
+        @test_warn ("INFO: Creating Example branch pinned.b1990792.tmp",
+                    "INFO: No packages to install, update or remove") Pkg.pin("Example", v"0.4.0")
         @test Pkg.installed("Example") == v"0.4.0"
 
-        ret, out, err = @grab_outputs Pkg.pin("Example", v"0.4.0")
-        @test ret === nothing && out == ""
-        @test contains(err, "INFO: Package Example is already pinned to the selected commit")
-        @test !contains(err, "INFO: No packages to install, update or remove")
```

Comment:
Wasn't deleted: added `r"^....$"` as above to make sure only one warning is printed